### DISCUSSION TITLE
Colab notebook stopped working (May 22, 2024)

### DISCUSSION THREAD

**Discussion by vitalybellen** [TheLastBen/fast-stable-diffusion#2837] [Q&A]
Hello! I apologize for creating this topic - I'm using your notebook for Google Colab (Pro). Everything was great, but today it stopped working and I encountered this error:

> WARNING[XFORMERS]: xFormers can't load C++/CUDA extensions. xFormers was built for:
>     PyTorch 2.2.1+cu121 with CUDA 1201 (you have 2.3.0+cu121)
>     Python  3.10.13 (you have 3.10.12)
>   Please reinstall xformers (see https://github.com/facebookresearch/xformers#installing-xformers)
>   Memory-efficient attention, SwiGLU, sparse and more won't be available.
>   Set XFORMERS_MORE_DETAILS=1 for more details

It might be something on my end, but reinstalling didn't help.
Also, when generating in Colab itself, this appears:

> NotImplementedError: No operator found for `memory_efficient_attention_forward` with inputs: query : shape=(1, 4096, 1, 512) (torch.float16) key : shape=(1, 4096, 1, 512) (torch.float16) value : shape=(1, 4096, 1, 512) (torch.float16) attn_bias : <class 'NoneType'> p : 0.0 `decoderF` is not supported because: max(query.shape[-1] != value.shape[-1]) > 128 xFormers wasn't build with CUDA support attn_bias type is <class 'NoneType'> operator wasn't built - see `python -m xformers.info` for more info `flshattF@0.0.0` is not supported because: max(query.shape[-1] != value.shape[-1]) > 256 xFormers wasn't build with CUDA support requires device with capability > (8, 0) but your GPU has capability (7, 5) (too old) operator wasn't built - see `python -m xformers.info` for more info `cutlassF` is not supported because: xFormers wasn't build with CUDA support operator wasn't built - see `python -m xformers.info` for more info `smallkF` is not supported because: max(query.shape[-1] != value.shape[-1]) > 32 xFormers wasn't build with CUDA support dtype=torch.float16 (supported: {torch.float32}) operator wasn't built - see `python -m xformers.info` for more info unsupported embed per head: 512

Thank you very much in advance!

upd: Adding a line before the execution cell helped me:
> !pip install -U xformers --indx-url [https://download.pytorch.org/whl/cu121](https://www.google.com/url?q=https%3A%2F%2Fdownload.pytorch.org%2Fwhl%2Fcu121)

**Comment by TheLastBen** [comment]
fixed
